cloudflare-template.sh

```diff
@@ -17,17 +17,26 @@ slackuri=""                                        # URI for Slack WebHook "http
 ###########################################
 ## Check if we have a public IP
 ###########################################
-ip=$(curl -s https://api.ipify.org || curl -s https://ipv4.icanhazip.com/)
+ipv4_regex='([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])'
+ip=$(curl -s -4 https://cloudflare.com/cdn-cgi/trace | grep -E '^ip'); ret=$?
+if [[ ! $ret == 0 ]]; then # In the case that cloudflare failed to return an ip.
+    # Attempt to get the ip from other websites.
+    ip=$(curl -s https://api.ipify.org || curl -s https://ipv4.icanhazip.com)
+else
+    # Extract just the ip from the ip line from cloudflare.
+    ip=$(echo $ip | sed -E "s/^ip=($ipv4_regex)$/\1/")
+fi
 
-if [ "${ip}" == "" ]; then 
-  logger -s "DDNS Updater: No public IP found"
-  exit 1
+# Use regex to check for proper IPv4 format.
+if [[ ! $ip =~ $ipv4_regex ]]; then
```

Comment:
This dosn't work, it an strange one but here simple test and soloution 
```ipv4_regex () {
  ip="$1"
  echo -n "IP [$ip] "
  if [[ ! $ip =~ ^$ipv4_regex$ ]]; then
    echo -n "[1] "
  fi
  if [[ ! $ip =~ $ipv4_regex ]]; then
    echo -n "[2] "
  fi
  echo " "
}

ipv4_regex='([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])'

echo "My test triggered [1] your test has triggered [2]"
ipv4_regex 0.0.0.0
ipv4_regex 1.2.3.4
ipv4_regex 012.004.000.199
ipv4_regex 255.255.255.255
ipv4_regex 0.0.0.256
ipv4_regex 100.100.100.256
ipv4_regex 100.100.100.356
ipv4_regex 0.0.256.0
ipv4_regex 255.255.255.256
ipv4_regex 256.256.256.256
ipv4_regex 0.0.300.0
ipv4_regex 0..2.3
ipv4_regex 0.1
ipv4_regex 0.1f.2
ipv4_regex computer```

Reply:
Thank you for catching that! I think I just forgot to add the `^` and `$` in this regex as it should be there. Strange that it doesn't work, but yes it should be there. I implemented this change in the last commit.